### PR TITLE
V08: Remove MessagePack Value from MessagePackParser

### PR DESCRIPTION
**Pull request by komamitsu**

### CODE REVIEW

File: msgpack-jackson/src/main/java/org/msgpack/jackson/dataformat/MessagePackParser.java [hunk 1]
```diff
@@ -199,9 +212,36 @@ public JsonToken nextToken()
                 }
                 break;
             case INTEGER:
-                value = messageUnpacker.unpackValue(var);
+                Object v;
+                switch (format) {
+                    case UINT64:
+                        BigInteger bi = messageUnpacker.unpackBigInteger();
+                        if (0 <= bi.compareTo(LONG_MIN) && bi.compareTo(LONG_MAX) <= 0) {
```
Comment (xerial) [inline msgpack-jackson/src/main/java/org/msgpack/jackson/dataformat/MessagePackParser.java]: Why not 'bi.compareTo(LONG_MIN) >=0`?

File: msgpack-jackson/src/main/java/org/msgpack/jackson/dataformat/MessagePackParser.java [hunk 2]
```diff
@@ -28,7 +28,7 @@
 import com.fasterxml.jackson.core.io.IOContext;
 import com.fasterxml.jackson.core.json.DupDetector;
 import com.fasterxml.jackson.core.json.JsonReadContext;
-import org.msgpack.core.MessageUnpacker;
+import org.msgpack.core.*;
```
Comment (xerial) [inline msgpack-jackson/src/main/java/org/msgpack/jackson/dataformat/MessagePackParser.java]: Expand this for vim users :)

File: msgpack-jackson/src/main/java/org/msgpack/jackson/dataformat/MessagePackParser.java [hunk 3]
```diff
@@ -209,9 +249,10 @@ public JsonToken nextToken()
                 }
                 break;
             case FLOAT:
-                value = messageUnpacker.unpackValue(var);
+                type = Type.DOUBLE;
+                doubleValue = messageUnpacker.unpackDouble();
```
Comment (xerial) [inline msgpack-jackson/src/main/java/org/msgpack/jackson/dataformat/MessagePackParser.java]: float to double conversion has some overhead. If you want to make it faster, consider to split reading float/double values separately (similarly int/long) in future.

File: msgpack-jackson/src/main/java/org/msgpack/jackson/dataformat/MessagePackParser.java (hunk 1 above)
Comment (komamitsu) [inline msgpack-jackson/src/main/java/org/msgpack/jackson/dataformat/MessagePackParser.java]: Just copied https://github.com/msgpack/msgpack-java/blob/a8f596dc0201195df361cfa819fabbdb52a23256/msgpack-core/src/main/java/org/msgpack/value/Variable.java#L452

File: msgpack-jackson/src/main/java/org/msgpack/jackson/dataformat/MessagePackParser.java (hunk 2 above)
Comment (komamitsu) [inline msgpack-jackson/src/main/java/org/msgpack/jackson/dataformat/MessagePackParser.java]: Okay fixed it.

File: msgpack-jackson/src/main/java/org/msgpack/jackson/dataformat/MessagePackParser.java (hunk 3 above)
Comment (komamitsu) [inline msgpack-jackson/src/main/java/org/msgpack/jackson/dataformat/MessagePackParser.java]: I see. Thanks.

**Comment by xerial** [comment]
@komamitsu Could you re-create a PR again by specifying v07-develop as the target branch? Thanks

**Comment by xerial** [comment]
Close this PR since v08-buffer-api is merged to v07-develop

**Pull request closed without merging**